overlord/snapstate/snapmgr.go

```diff
@@ -1006,37 +997,31 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	if err != nil {
 		return err
 	}
-
-	isRevert := false
-	oldCandidateIndex := -1
-	if ss.HasABA() {
-		isRevert = ss.IsRevert()
-		err = t.Get("old-candidate-index", &oldCandidateIndex)
-		if err != nil {
-			return err
-		}
-	} else {
-		err = t.Get("had-candidate", &isRevert)
-		if err != nil && err != state.ErrNoState {
-			return err
-		}
+	var oldCandidateIndex int
+	if err := t.Get("old-candidate-index", &oldCandidateIndex); err != nil {
+		return err
 	}
 
+	isRevert := ss.Flags.Revert()
+
 	// relinking of the old snap is done in the undo of unlink-current-snap
 	currentIndex := snapst.LastIndex(snapst.Current)
 	if currentIndex < 0 {
 		return fmt.Errorf("internal error: cannot find revision %d in %v for undoing the added revision", ss.SideInfo.Revision, snapst.Sequence)
 	}
-	if ss.HasABA() {
-		if oldCandidateIndex < 0 {
-			snapst.Sequence = append(snapst.Sequence[:currentIndex], snapst.Sequence[currentIndex+1:]...)
-		} else if !isRevert {
-			oldCand := snapst.Sequence[currentIndex]
-			copy(snapst.Sequence[oldCandidateIndex+1:], snapst.Sequence[oldCandidateIndex:])
-			snapst.Sequence[oldCandidateIndex] = oldCand
-		}
-	} else if !isRevert {
+
+	if oldCandidateIndex >= 0 && !isRevert {
+		oldCand := snapst.Sequence[currentIndex]
+		copy(snapst.Sequence[oldCandidateIndex+1:], snapst.Sequence[oldCandidateIndex:])
+		snapst.Sequence[oldCandidateIndex] = oldCand
+	}
```

Comment:
this and below seem to do the same thing twice? and is that correct? no test is exploding?